articles/cms/wordpress/extending.md

```diff
@@ -0,0 +1,230 @@
+}
+add_action( 'auth0_before_login', 'a0_docs_ex_auth0_before_login' );
+``` 
+
+### `auth0_user_login` 
+
+This action runs in `WP_Auth0_LoginManager` after a user has been authenticated successfully and logged into 
+WordPress. It can be used to set specific meta values, send notifications, or ping other services. 
+
+**Example:**
+
+```php
+/**
+ * @param integer $user_id - WordPress user ID
+ * @param stdClass $userinfo - user information object from Auth0
+ * @param boolean $is_new - true if the user was created in WordPress, false if not
+ * @param string $id_token - ID token for the user from Auth0 (not used in code flow)
+ * @param string $access_token - bearer access token from Auth0 (not used in implicit flow)
+ */
+function a0_docs_ex_auth0_user_login ( $user_id, $userinfo, $is_new, $id_token, $access_token ) {
+  // ... do something after logging in
+}
+add_action( 'auth0_user_login', 'a0_docs_ex_auth0_user_login', 10, 5 );
+``` 
+
+### `wpa0_user_created` 
+
+This action runs in `WP_Auth0_Users` just after a WordPress user is successfully created. It can be used to change 
+user values, set additional user metas, or trigger other new user actions. 
+
+**Example:**
+
+```php
+/**
+ * @param integer $user_id - WordPress user ID for created user
+ * @param string $email - email address for created user
+ * @param string $password - password used for created user
+ * @param string $f_name - first name for created user
+ * @param string $l_name - last name for created user
+ */
+function a0_docs_ex_wpa0_user_created ( $user_id, $email, $password, $f_name, $l_name ) {
+	// ... do something once a user has been created
+}
+
+add_action( 'wpa0_user_created', 'a0_docs_ex_wpa0_user_created', 10, 5 );
+```
+
+## Filters
+
+Filters in WordPress also run custom code at specific points during processing but always return a modified value of the same type that was passed in. [Learn more about filters here](https://developer.wordpress.org/plugins/hooks/filters/). 
+
+### `auth0_get_wp_user`
+
+This filter is called after the plugin finds the related user to login (based on the auth0 `user_id`) and is used to override the default behaviour with custom matching rules (for example, always match by email).
+
+If the filter returns null, it will lookup by email as described in the [How does it work?](https://auth0.com/docs/cms/wordpress/how-does-it-work) document.
+
+**Example:**
+
+```php
+/**
+ * @param WP_User|null $user - found WordPress user, null if no user was found
+ * @param stdClass $userinfo - user information from Auth0
+ *
+ * @return WP_User|null
+ */
+function auth0_theme_hook_auth0_get_wp_user( $user, $userinfo ) {
+	// ... try to match another way, decide to replace $user
+	return $user;
+}
+add_filter( 'auth0_get_wp_user', 'auth0_theme_hook_auth0_get_wp_user', 1, 2 );
+```
+
+### `auth0_verify_email_page`
+
+This filter runs in `WP_Auth0_Email_Verification` to change the HTML rendered when a logging-in user needs to verify 
```

Comment:
...when a user who is logging in needs to...

Reply:
Jet lag grammer 😆